protocols/kad/src/query.rs

```diff
@@ -143,9 +143,9 @@ impl QueryPool {
     {
         let num_results = match info {
             QueryInfo::GetClosestPeers {
-                num_results: Some(val),
-                ..
+                num_results: val, ..
             } => val,
+            QueryInfo::Bootstrap { .. } => K_VALUE,
```

Comment:
During `Bootstrap` queries, we expect `K_VALUE` results (and not `replication_factor`)